tfe/resource_tfe_policy_set.go

```diff
@@ -0,0 +1,300 @@
+package tfe
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceTFEPolicySet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEPolicySetCreate,
+		Read:   resourceTFEPolicySetRead,
+		Update: resourceTFEPolicySetUpdate,
+		Delete: resourceTFEPolicySetDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceTFEPolicySetImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"global": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"organization": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"policies": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"workspaces": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceTFEPolicySetCreate(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	global := d.Get("global").(bool)
+	organization := d.Get("organization").(string)
+	policies := d.Get("policies").(*schema.Set)
+	workspaces := d.Get("workspaces").(*schema.Set)
+
+	// First off, make sure they aren't trying to supply workspaces for a global set
+	if global && workspaces.Len() > 0 {
+		err := errors.New(`cannot set "workspaces" when "global" is true`)
+		return fmt.Errorf(
+			"Error creating policy set %s for organization %s: %v", name, organization, err)
+	}
+
+	// Create a new options struct.
+	options := tfe.PolicySetCreateOptions{
+		Name:        tfe.String(name),
+		Description: tfe.String(description),
+		Global:      tfe.Bool(global),
+	}
+
+	// Set up the policies and workspaces.
+	for _, policyID := range policies.List() {
+		options.Policies = append(options.Policies, &tfe.Policy{ID: policyID.(string)})
+	}
+
+	if !*options.Global {
```

Comment:
Since you have `global` defined already, why not reuse that here to make it read just a bit nicer? But to be honest I think we don't need this check at all, as we already checked in the beginning of this function if `global` was `true` and `workspaces` were defined. So the `range` op should be a no-op here.